src/commonMain/kotlin/io/modelcontextprotocol/kotlin/sdk/server/StreamableHttpServerTransport.kt

```diff
@@ -0,0 +1,304 @@
+                correspondingCall.response.headers.append(ContentType.toString(), ContentType.Application.Json.toString())
+                correspondingCall.response.status(HttpStatusCode.OK)
+                if (sessionId != null) {
+                    correspondingCall.response.header("Mcp-Session-Id", sessionId!!)
+                }
+                val responses = relatedIds.map{ requestResponseMapping[it] }
+                if (responses.size == 1) {
+                    correspondingCall.respond(responses[0]!!)
+                } else {
+                    correspondingCall.respond(responses)
+                }
+                callMapping.remove(streamId)
+            } else {
+                correspondingStream.close()
+                streamMapping.remove(streamId)
+            }
+
+            for (id in relatedIds) {
+                requestToStreamMapping.remove(id)
+                requestResponseMapping.remove(id)
+            }
+        }
+
+    }
+
+    override suspend fun close() {
+        streamMapping.values.forEach {
+            it.close()
+        }
+        streamMapping.clear()
+        requestToStreamMapping.clear()
+        requestResponseMapping.clear()
+        // TODO Check if we need to clear the callMapping or if call timeout after awhile
+        _onClose.invoke()
+    }
+
+    @OptIn(ExperimentalUuidApi::class)
+    public suspend fun handlePostRequest(call: ApplicationCall, session: ServerSSESession) {
+        try {
+            val acceptHeader = call.request.headers["Accept"]?.split(",") ?: listOf()
+
+            if (!acceptHeader.contains("text/event-stream") || !acceptHeader.contains("application/json")) {
+                call.response.status(HttpStatusCode.NotAcceptable)
+                call.respond(
+                    JSONRPCResponse(
+                        id = null,
+                        error = JSONRPCError(
+                            code = ErrorCode.Unknown(-32000),
+                            message = "Not Acceptable: Client must accept both application/json and text/event-stream"
+                        )
+                    )
+                )
+                return
+            }
+
+            val contentType = call.request.contentType()
+            if (contentType != ContentType.Application.Json) {
+                call.response.status(HttpStatusCode.UnsupportedMediaType)
+                call.respond(
+                    JSONRPCResponse(
+                        id = null,
+                        error = JSONRPCError(
+                            code = ErrorCode.Unknown(-32000),
+                            message = "Unsupported Media Type: Content-Type must be application/json"
+                        )
+                    )
+                )
+                return
+            }
+
+            val body = call.receiveText()
+            val messages = mutableListOf<JSONRPCMessage>()
+
+            if (body.startsWith("[")) {
```

Comment:
I would prefer to receive a JsonElement and check if the item is an array. Otherwise, it will break when a space or a new line is used